src/Mafia/Twine/Async.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Mafia.Twine.Async (
+    AsyncTimeout (..)
+  , renderAsyncTimeout
+  , waitWithTimeout
+  , waitEitherBoth
+  ) where
+
+import           Control.Concurrent.Async (Async, waitSTM, waitEither)
+import           Control.Concurrent.Async (async, cancel, wait)
+import           Control.Concurrent.STM (atomically, orElse, retry)
+import           Control.Exception.Base (AsyncException (..))
+import           Control.Monad.Catch (catch, throwM)
+
+import           Control.Monad.IO.Class (liftIO)
+
+import           Data.IORef (newIORef, readIORef, writeIORef)
+import qualified Data.Text as T
+
+import           Mafia.P
+
+import           Mafia.Twine.Snooze
+
+import           System.IO (IO)
+
+import           Control.Monad.Trans.Either
+
+data AsyncTimeout =
+  AsyncTimeout Duration
+  deriving (Eq, Show)
```

Comment:
Any reason for that not to be a `newtype`?

Reply:
I'm purposefully not improving / changing twine beyond what is absolutely necessary to get mafia working. There's a good case for either making twine a real thing on hackage or improving the twine inlined here (I'd be voting for the first option).